### ISSUE TITLE
[Component Update] Evolution on display of error message and suffix/prefix for text input

### ISSUE THREAD

**Issue by ludovic35** [Orange-OpenSource/ouds-ios#1063]
### Description

Two main evolutions: Error message is mandatory if on error, and display prefix/suffix in focus state even if placeholder is empty

**1. When the status is set on error an error message is mandatory and helper text is ignored.**

Proposition add a string in the error status

```
    public enum Status {
        case enabled
        case error(message: String)
        case loading
...
```

Set in documentation that the helper text is ignored in that case.
Reflect this evolution in design tool box to update the textfield of the helper text to error text field

**2. Suffix and prefix should be display event if placeholder is empty in focus state**

### Types of change

- [ ] Add new component
- [x] Update existing component

### Checklist for Core Team

- [ ] Tokens have been updated
- [x] Documentations have been updated
- [ ] Unit tests have been updated
- [x] UI tests have been updated
- [x] Design System Toolbox has been updated
- [ ] Retrocompatibility has been checked, or broken API have been flagged as deprecated with a workaround

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **0.21.0**
- Build number: **1762207613**
- Build tag: **0bd5fb8**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_
